src/all/pandachaika/src/eu/kanade/tachiyomi/extension/all/pandachaika/PandaChaika.kt

```diff
@@ -0,0 +1,326 @@
+                .joinToString {
+                    it.substringAfter(":").replace("_", " ").split(" ").joinToString(" ") { s ->
+                        s.replaceFirstChar { sr ->
+                            if (sr.isLowerCase()) sr.titlecase(Locale.getDefault()) else sr.toString()
+                        }
+                    }
+                }
+        }
+        fun getReadableSize(bytes: Double): String {
+            return when {
+                bytes >= 300 * 1024 * 1024 -> "${"%.2f".format(bytes / (1024.0 * 1024.0 * 1024.0))} GB"
+                bytes >= 100 * 1024 -> "${"%.2f".format(bytes / (1024.0 * 1024.0))} MB"
+                bytes >= 1024 -> "${"%.2f".format(bytes / (1024.0))} KB"
+                else -> "$bytes B"
+            }
+        }
+        val groups = filterTags("group")
+        val artists = filterTags("artist")
+        val publishers = filterTags("publisher")
+        val male = filterTags("male")
+        val female = filterTags("female")
+        val others = filterTags(exclude = listOf("female", "male", "artist", "publisher", "group", "parody"))
+        val parodies = filterTags("parody")
+        title = this@toSManga.title
+        url = download.substringBefore("/download/")
+        author = groups.ifEmpty { artists }
+        artist = artists
+        genre = listOf(male, female, others).joinToString()
+        description = buildString {
+            append("Uploader: ", uploader, "\n")
+            publishers.takeIf { it.isNotBlank() }?.let {
+                append("Publishers: ", it, "\n\n")
+            }
+            parodies.takeIf { it.isNotBlank() }?.let {
+                append("Parodies: ", it, "\n\n")
+            }
+            male.takeIf { it.isNotBlank() }?.let {
+                append("Male tags: ", it, "\n\n")
+            }
+            female.takeIf { it.isNotBlank() }?.let {
+                append("Female tags: ", it, "\n\n")
+            }
+            others.takeIf { it.isNotBlank() }?.let {
+                append("Other tags: ", it, "\n\n")
+            }
+
+            title_jpn?.let { append("Japanese Title: ", it, "\n") }
+            append("Pages: ", filecount, "\n")
+            append("File Size: ", getReadableSize(filesize), "\n")
+
+            try {
+                append("Posted: ", dateReformat.format(Date(posted * 1000)), "\n")
+            } catch (_: Exception) {}
+        }
+        status = SManga.COMPLETED
+        update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+        initialized = true
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        return response.parseAs<Archive>().toSManga()
+    }
+
+    // Chapters
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val archive = response.parseAs<Archive>()
+
+        return listOf(
+            SChapter.create().apply {
+                name = "Chapter"
+                url = archive.download.substringBefore("/download/")
```

Comment:
can directly save download url and start parsing the zip directly instead of fetching again in page list parsing. (override fetchPageList for that)